controllers/apim/authpolicy_controller.go

```diff
@@ -131,22 +131,19 @@ func (r *AuthPolicyReconciler) reconcileAuthSchemes(ctx context.Context, ap *api
 func (r *AuthPolicyReconciler) reconcileAuthRules(ctx context.Context, ap *apimv1alpha1.AuthPolicy) error {
 	logger, _ := logr.FromContext(ctx)
 
-	httpRoute, err := r.fetchHTTPRoute(ctx, ap)
+	targetObj, err := r.fetchTargetObject(ctx, ap)
+	targetObjectKind := targetObj.GetObjectKind().GroupVersionKind().Kind
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			logger.Info("referenced HTTPRoute not found")
+			logger.Info("referenced %s not found", targetObjectKind)
 			return nil
 		}
 		return err
 	}
 
-	for _, parentRef := range httpRoute.Spec.ParentRefs {
-		gwNamespace := httpRoute.Namespace // consider gateway local if namespace is not given
-		if parentRef.Namespace != nil {
-			gwNamespace = string(*parentRef.Namespace)
-		}
-		gwName := string(parentRef.Name)
+	gwKeys := TargetedGatewayKeys(targetObj)
```

Comment:
Idea for a possible refactoring in the future:

I wonder whether we couldn't have a `TargetObject` interface – since [sigs.k8s.io/gateway-api/apis](https://pkg.go.dev/sigs.k8s.io/gateway-api/apis) seems to provide nothing like that.

Maybe starting w/ sth like:

```go
type TargetObject interface {
  client.Object
  func GetKind() gatewayapiv1alpha2.Kind
  func TargetedGatewayKeys() []client.ObjectKey
}
```

And then concrete types for `HTTPRoute`, `Gateway`, etc.

WDYT?

Reply:
I have added this to my TODO so we can merge this PR for now.

Reply:
We'd need concrete types of our own that hold references to actual `HTTPRoute` and `Gateway` objects, but implement the interface.

Reply:
hmm, okay, got it. will do in the next PR. :)